tools/c7n_org/c7n_org/cli.py

```diff
@@ -458,6 +458,599 @@ def report(config, output, use, output_dir, accounts,
+    """Run per-account validation with variable expansion.
+
+    Validates policies for each account, expanding account-specific variables
+    and checking for missing or invalid variable references.
+
+    This function validates that policies work correctly with each account's
+    specific configuration and variables. It handles multi-cloud scenarios by
+    matching policy providers with account providers.
+
+    Args:
+        custodian_config: Parsed policy configuration
+        accounts_config: Parsed account configuration
+        policy_file: Path to policy file (for error reporting)
+        fmt: File format ('yml', 'yaml', 'json')
+        check_mode: Deprecation check mode
+        verbose: Verbose logging flag
+
+    Returns:
+        bool: True if validation passes for all accounts, False otherwise
+    """
+    import copy
+    from c7n import deprecated
+    from c7n.policy import Policy
+    from c7n.config import Config, Bag
+    from c7n.loader import SourceLocator
+
+    accounts = accounts_config['accounts']
+    policies = custodian_config.get('policies', [])
+
+    log.info(f"Validating {len(policies)} policies across {len(accounts)} accounts")
+
+    # First, run basic validation to catch structural issues
+    log.debug("Running initial basic validation")
+    if not validate_basic(custodian_config, policy_file, fmt, check_mode, verbose):
+        return False
+
+    # Track results per account
+    account_results = {}
+    overall_success = True
+
+    # Validate for each account
+    for account in accounts:
+        account_name = account.get('name', account.get('account_id', 'unknown'))
+        log.info(f"Validating for account: {account_name}")
+
+        account_errors = []
+        account_warnings = []
+
+        # Get account-specific variables
+        account_vars = account.get('vars', {})
+        if verbose and account_vars:
+            log.debug(f"  Account variables: {list(account_vars.keys())}")
+
+        # Get account info (already normalized by accounts_iterator)
+        account_id = account.get('account_id', 'na')
+        account_provider = account.get('provider', 'aws')
+
+        # Use region from account config
+        # accounts_iterator() has already set appropriate region defaults per provider
+        regions = account.get('regions', ['na'])
+        region = regions[0] if regions else 'na'
+
+        if verbose:
+            log.debug(f"  Account provider: {account_provider}, region: {region}")
+
+        # Create config for this account
+        account_config = Config.empty(
+            dryrun=True,
+            account_id=account_id,
+            region=region
+        )
+
+        source_locator = None
+        if fmt in ('yml', 'yaml'):
+            source_locator = SourceLocator(policy_file)
+
+        # Validate each policy with account context
+        for policy_data in policies:
+            policy_name = policy_data.get('name', 'unknown')
```

Comment:
_nit:_ This is a pretty big `for` loop in a pretty big function. If there's any low-hanging fruit, maybe extraction of some of the logic?